pkg/utils/common.go

```diff
@@ -125,23 +125,28 @@ func ValidateNamespace(ns string) error {
 // user inputs that might be provided.
 func SanitizeRepoUrl(url string) string {
 	trimmed := ""
+	domain := "github"
+
+	if strings.Contains(url, "gitlab") {
```

Comment:
This will be a much bigger restructure. We cannot use normalizedRepoURL as a parameter because gitproviders imports common. We also sanitize the url in a bunch of random places.

Reply:
@J-Thompson12 Can you skip `utils` all together? Like, whatever is consuming this `SanitizeRepoURL` func can just use `NormalizedRepoURL`>